### DISCUSSION TITLE
Releases API sometimes is not reliable

### DISCUSSION THREAD

**Discussion by sophia-guo** [community/community#24725] [Apps, API and Webhooks]
<p>Releases API sometimes works and sometimes doesn’t. Looks like it really depends on how the releases is generated? I’ve met quite a few repos the API doesn’t work. Is there any reason why and how it can be fixed?</p>
<p>For example the following query return nothing<br>
curl -H “Accept: application/vnd.github.v3+json” <a href="https://api.github.com/repos/openjdk/jtreg/releases" rel="noopener nofollow ugc">https://api.github.com/repos/openjdk/jtreg/releases</a></p>
<p>While different repos works:<br>
curl -H “Accept: application/vnd.github.v3+json” <a href="https://api.github.com/repos/adoptium/run-aqa/releases" rel="noopener nofollow ugc">https://api.github.com/repos/adoptium/run-aqa/releases</a></p>

**Comment by shrink** [comment, marked as answer]
<p>The GitHub Repository “Releases” page is not <em>just</em> Releases, the interface co-mingles Releases (something you create on GitHub) and Tags (a native git feature). A Tag points to a specific commit and a Release, in turn, points to a Tag, but they’re co-mingled on the interface because for a consumer it is usually the Tag that matters for <em>both</em> a Tag and a Release.</p>
<p>If we visit the <a href="https://github.com/openjdk/jtreg/releases" rel="noopener nofollow ugc">Releases of openjdk/jtreg</a> we can see that there are many Tags, but no Releases, whereas in the <a href="https://github.com/adoptium/run-aqa/releases" rel="noopener nofollow ugc">Releases of adoptium/run-aqa</a> we can see there are many Releases.</p>
<p>The difference between a Release and a Tag on the interface can be identified based on the <em>style</em> of information present, and the information included. A Release has a large heading, and a sub-heading that reads “<span class="mention">@username</span> released this n time ago” whereas a Tag simply displays the Tag and the commit it points to.</p>
<h2>
<a name="release-1" class="anchor" href="#release-1"></a>Release</h2>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/0/9/09a9da6d2bb8bd8542d21e0313d0c1768342dc0b.png" data-download-href="/uploads/short-url/1nucMeM3ssMkNEmsdCfse82pr3d.png?dl=1" title="Screenshot 2021-07-16 at 22.34.05" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/349689/181086467-ff730948-ba73-4a1b-b931-bc5d37776443.png" alt="Screenshot 2021-07-16 at 22.34.05" data-base62-sha1="1nucMeM3ssMkNEmsdCfse82pr3d" width="690" height="158" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/0/9/09a9da6d2bb8bd8542d21e0313d0c1768342dc0b_2_690x158.png, https://aws1.discourse-cdn.com/github/optimized/3X/0/9/09a9da6d2bb8bd8542d21e0313d0c1768342dc0b_2_1035x237.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/0/9/09a9da6d2bb8bd8542d21e0313d0c1768342dc0b_2_1380x316.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/0/9/09a9da6d2bb8bd8542d21e0313d0c1768342dc0b_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot 2021-07-16 at 22.34.05</span><span class="informations">2440×560 38.6 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<h2>
<a name="tag-2" class="anchor" href="#tag-2"></a>Tag</h2>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/4/7/478f1dac8a710f50c989ce1863aacdbecd2244ea.png" data-download-href="/uploads/short-url/ad2wxswqWUINZ80ToK0OrL10vLs.png?dl=1" title="Screenshot 2021-07-16 at 22.34.22" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/349689/181086468-8d4996e8-ff6a-4144-9fee-9be4c7136fcd.png" alt="Screenshot 2021-07-16 at 22.34.22" data-base62-sha1="ad2wxswqWUINZ80ToK0OrL10vLs" width="690" height="45" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/4/7/478f1dac8a710f50c989ce1863aacdbecd2244ea_2_690x45.png, https://aws1.discourse-cdn.com/github/optimized/3X/4/7/478f1dac8a710f50c989ce1863aacdbecd2244ea_2_1035x67.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/4/7/478f1dac8a710f50c989ce1863aacdbecd2244ea_2_1380x90.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/4/7/478f1dac8a710f50c989ce1863aacdbecd2244ea_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot 2021-07-16 at 22.34.22</span><span class="informations">2440×160 11.2 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>If you’d like to programmatically access a list of tags, there is an endpoint, just <a href="https://docs.github.com/en/rest/reference/repos#list-repository-tags" rel="noopener nofollow ugc">replace <code>releases</code> with <code>tags</code></a>.</p>
<pre><code class="lang-auto">GET https://api.github.com/repos/openjdk/jtreg/tags
</code></pre>
<p>You may also find <a href="https://docs.github.com/en/rest/reference/repos#get-a-release-by-tag-name" rel="noopener nofollow ugc">get a release by tag name</a> helpful.</p>

**Comment by sophia-guo** [comment]
<p>Thanks <a class="mention" href="/shrink">@shrink</a> ! That makes sense. Unfortunately using the tag instead of releases can’t  assume the tags are in chronological order( in alphabetical order I believe), which makes it difficult to get the latest tag for some repo.</p>

**Comment by sophia-guo** [comment]
<p>Will use git command <code>git tag --sort=taggerdate</code> with checkout the repository. Close this one.</p>
